src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -665,13 +672,13 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
                         Error(diagnostics, ErrorCode.ERR_AnonymousTypeDuplicatePropertyName, argumentSyntax.NameColon.Name);
```

Comment:
In your example the error after the change is about a new name and we are not in the business of predicting how user is going to change the name.